tutor/src/screens/gradebook/settings-btn.js

```diff
@@ -0,0 +1,86 @@
+import { React, PropTypes, useState, useRef, useObserver, styled } from 'vendor';
+import { Button, Popover, Overlay } from 'react-bootstrap';
+import { Icon } from 'shared';
+import UX from './ux';
+
+const Toggles = styled(Popover.Content)`
+  padding: 1rem;
+  label {
+    display: block;
+  }
+  input {
+    margin-right: 1rem;
+  }
+`;
+
+const SettingsCheckbox = ({ ux, title, property }) => useObserver(() => (
+  <label>
+    <input
+      type="checkbox"
+      checked={ux[property]}
+      onChange={({ target }) => ux[property] = target.checked}
+    />
+    {title}
+  </label>
+));
+  
+SettingsCheckbox.propTypes ={
+  ux: PropTypes.instanceOf(UX).isRequired,
+  title: PropTypes.string.isRequired,
+  property: PropTypes.string.isRequired,
+};
+  
+const Settings = ({ ux }) => {
+  const [showPopover, setShowPopover] = useState(false);
+  const [showSettings, setShowSettings] = useState(false);
```

Comment:
We usually put things like this inside a UX class and use mobx observables.

Reply:
:+1:

Reply:
@nathanstitt @jivey actually, instead of putting this open and hide popovers/modals, is it ok to put the logic in the same class using observables? same as I am doing in `export-btn` and `lms-push-btn`.

Reply:
Yeah, if the behavior is localized to the component, that makes sense to me. Though I think you'd have to switch it to a class component.

Reply:
Yeah, I'm totally onboard with converting to class component.  this was kinda an experiment to see how functional would look, and so far I slightly prefer class component, as it reads clearer to me

Reply:
Thank you guys. I updated the code based on the comments. One more :eye: :eye: